core/src/scene/dataLayer.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "scene/sceneLayer.h"
+#include <string>
+
+namespace Tangram {
+
+// DataLayer represents a top-level layer in the stylesheet, distinct from
+// SceneLayer by its association with a collection within a DataSource
+class DataLayer {
```

Comment:
I think it will simplify many things... like how DataLayer->SceneLayers are constructed when yaml is parsed. It will also make access like [this](https://github.com/tangrams/tangram-es/pull/237/files#diff-25abeab428d409be23fa79dd35450968R60), more direct.

And logic wise it seems to be more of a inheritance model rather than a containership model.

(Also my battery running low now :|, and charging slot is occupied by my co-passenger :().